### ISSUE TITLE
Elementor 1.9 - Beta Release (RC3 Updated)

### ISSUE THREAD

**Issue by bainternet** [pojome/elementor#3067]
===========
## [RC3 UPDATED - PLEASE READ BELOW!](https://github.com/pojome/elementor/issues/3067#issuecomment-355981568)
===========

Hi Everyone,

2018 is here, and boy, do we have some updates to blast your workflow into the future!

Elementor version 1.9 comes packed with tons of new features and redesigns.  
Now’s your chance to explore everything that’s new, making sure it’s all in tip-top shape, ready to be shipped to everyone.

**So what’s new?**  

We have a long list of changes, including Total Template Library makeover, new Autosave, new Draft / Publish statuses, Dashboard Widget and many other improvements ready for your testing hands.  

We have a lot to cover. Ladies and gentlemen... start your testing!

### Template Library Makeover

We have now redesigned Elementor’s Template Library, making it significantly easier to navigate and sort templates in terms of functionality as well as UI.  

Go ahead and test these new Library features:

**Template filter**  
Check out our brand new filtering system, which lets you sort new/old templates, popular template (downloads) and trendy templates (most downloads in recent days).

**Template search**  
The search box lets you search for a certain template or tag. Templates now have relevant tags, so you can also search for a certain category like ‘Sport’ or ‘Food’.

**Template favorites**  
The heart icon next to each template lets you save any template to the ‘My Favorites’ tab, for easy access to your most loved templates.

**Sort ‘My Templates’**  
Under the ‘My Templates’ tab, you can sort your templates according to: Name, Type (page/section), Author, Date. You can also filter your saved templates by using the search bar.

**Sync Library directly from the popup**  
We’ve added the ‘Sync Library’ button to the Library window. Previously, this button was only available under the tool's dashboard. Now you can synchronize your library straight from the Library popup.

**Import templates directly**  
We’ve also added the ‘Import Template’ functionality to the ‘My Templates’ tab, so you can import templates directly from the Library popup.

**Responsive Library window**  
The library popup window is now more responsive, making it a perfect fit for all devices.

### Introducing Autosave

You no longer need to remember to press save. The page is automatically saved five seconds after you make any change.  
Having autosave lets you focus more clearly on designing the page. With autosave, you never lose your work.

**How it works?**

*   Make a few changes to the page, drag a widget or change a color.
*   Wait a few seconds, and notice that the page gets automatically saved.
*   Open the saving button on the right of the footer panel to see the message: ‘Last edited on Jan 1, 15:54 by Authorname’.
*   Go to History > Revisions and see the autosave that was just created.

### Draft / Publish Statuses

On previous versions, you only had one version you could work on at any given moment - the live version of the page.  
Now, Elementor lets you work on drafts, similar to how you edit pages with the WordPress editor.

Once you’re done editing, click ‘Publish’ to instantly get the page live and published. You can also exit Elementor and keep the page as a draft.

When you edit a page that has already been published, all your new changes will only affect the draft, and not the live version. The changes will take effect only once you hit ‘Update’.

### Elementor Dashboard Widget

This version introduces a new Elementor widget into the WordPress Dashboard.  
Using this widget, you’ll be able to do all kinds of cool things like creating a new page with Elementor, accessing recently created documents and reading about newly added feature updates.

### Other Improvements

**NEW Editor Footer Panel:**  
- The ‘Save as template’ button has been moved inside the ‘Save Options’ button.  
- The ‘X’ exit button on the bottom left has been removed. You can now exit the page through the ‘Publish / Update’ button or through the main menu on the top left.  
- The ‘Page Settings’ menu has been renamed ‘Document Settings’, and has been moved to the bottom left of the footer for easier access.

**Popup Control:**  
If you edit any text widget, you will see under ‘Style’ that the Typography settings have now been changed to a popup window. The same is true to Box-shadow and Text-shadow settings. This change will help you get an easier workflow and focus when customizing style settings.

**More Privacy:**  
The Video widget now has a Privacy Mode. This way, you can make sure YouTube doesn’t save cookies in your visitors’ browsers.

**Gutenberg Compatibility:**  
- Under the Posts and Pages dashboard, you have a new option to open a new page with Elementor directly, not going first through Gutenberg or the classic editor.  
- When you edit a page with Elementor, and then go back to the dashboard, you are directed to the classic dashboard and not to the Gutenberg editor. This fixes a bug that Gutenberg users were experiencing.

**Dev Improvements:**

*   **Depended Styles for developers** - Now various elements can set stylesheet dependencies.
*   **Elementor switched date picker library** - The old `jquery-simple-dtpicker` was replaced with `flatpickr`. The new library fixes various iOS issues in Elementor controls that have a date or time pickers. It is also much lighter, which in turn is faster to load, and does not rely on jQuery.
*   **Inline Code Documentation** - We've started a huge ongoing project to document the entire code with phpDocs. Functions, action and filter hooks have already been documented, and the work continues for classes, methods, properties and so on.

### How to Install Beta Version

If you still haven't enabled beta testing in Elementor, please follow these steps:

1.  Go to Elementor > Tools > Versions
2.  Enable the beta tester feature
3.  Go to Dashboard > Updates and update to the latest beta version

This will get you testing the new Elementor 1.9.0 RC1\. You can also download the new version here:  
https://downloads.wordpress.org/plugin/elementor.1.9.0-RC3.zip

### Notes

**Designated time for release:** January 9, 2018

Many thanks for your support and help!

**For bugs and remarks, please reply with a comment below!**

[Elementor Beta - Sign Up Now For Our Elite Group Of Early Access Testers](https://my.elementor.com/beta/)

Cheers,

**Comment by norewp** [comment]
Been playing with this branch for a couple weeks now and everything is cool but............auto save, please turn it off and make it optional! An option to purge revisions also would be nice - otherwise, great job guys :)

**Comment by soyket** [comment]
Nice but we need autosave limit / purge option. It takes and keeps the data for infinite which is a problem for users with low disk space, db space etc.

Thanks

**Comment by bainternet** [comment]
@norewp @soyket 

I'll to be as clear as possible:
# AUTOSAVE DOESNT CREATE REVISIONS!!!

> The max number or revisions is still the same as you define using:
> 
> `define( 'WP_POST_REVISIONS', 10 );`
> 
> nothing about the revisions is changed!!
> 
> I'll try to explain how it works which is just like the native WordPress Editor,
> 
> When you start a new post then its automatically set as draft so any changes you make, is saved to that draft automatically. (no revisions yet).
> Once you "Publish" that post, then the draft is turned into published. (still no revisions).
> 
> When you edit a published post, on the first change you make we create a new "Auto-Draft" ( still just like the native WordPress editor )
> which is updated with every change you make ("Autosave"). Now once you decide to publish these changes (the "update" button ) then that "Auto-Draft"
> becomes the published version and only now the old published version is saved as a revision.

Basically, we matched the native WordPress Autosave/draft behavior.

**Comment by soyket** [comment]
Ah! I see I didn't check it that way! I guess it's okay then!

**Comment by norewp** [comment]
You should have said that in the first place @bainternet  - some of us have weak hearts, be kind please :P 

To most WordPress users even some of us seasoned devs (yes me too) "auto save" says revisions for some reason. Time to reshape the thinking mind :)

**Comment by anandau** [comment]
Where gone the option to open Elementor library from editor window? 
Not able to locate this old option in new UI

![t_lib](https://user-images.githubusercontent.com/376489/34492270-dc4a53b8-f00c-11e7-8a0e-0e8231658eea.png)

**Comment by Sarah-p** [comment]
the new update button is nice with "save & exit/ discard & exit/ save as template" BUT you lost the view page link. The preview one isn't the same nor is it really needed. When you minimize the widget panel you get a preview anyway so why not make the preview button go to the live page?
![e1 9rc](https://user-images.githubusercontent.com/29761063/34495107-c308e344-ef97-11e7-9b24-0d4cae78a3b2.jpg)

**Comment by tscheipi** [comment]
# Templates:
- nice features
-  what about showing own templates first by default?
- what about the ability to export more than one template at once?
- Responsive Library window doesn´t work here. making the window smaller there are allways 4 columns.
# Dashboard Widget:
- option to creat new page is great, but missing the option to create new posts!?
- help: if you made a site for clients you might want the client to contact you.
- my account: if a client doesn´t have an account this could be irritating?
# More Privacy:
- thumbs up! real great feature!
# Depended Styles for developers + Inline Code Documentation:
- sry, but don´t understand this
- where to find/ how to use?

erverything is working fine here except the responsive library window

**Comment by wsydney76** [comment]
Creating a new page and adding some headlines/editor (without clicking "publish") does indeed create a lot of revisions:

![image](https://user-images.githubusercontent.com/25308802/34499405-3cc2351e-f006-11e7-9711-7b4761576485.png)

![image](https://user-images.githubusercontent.com/25308802/34499592-0777f352-f007-11e7-8a9c-f7974006cdf4.png)

**Comment by wsydney76** [comment]
@anandau Glad i am not the only one who fell into that trap!

It's still in the content area:

![image](https://user-images.githubusercontent.com/25308802/34500257-3680c414-f00a-11e7-91e5-16488c32e932.png)

**Comment by bainternet** [comment]
@wsydney76 

Thanks for reporting I can confirm there is a bug when post status is set to draft (before publishing), revisions are created wrongly.

@tscheipi 

Thanks for the feedback, few answers

> what about showing own templates first by default?

 => follow this issue #2456 

> what about the ability to export more than one template at once? 

 => it's already in the backend since version  1.6 using bulk actions

> Responsive Library window doesn´t work here. making the window smaller there are always 4 columns.

 => Thanks, will look at it.

> option to create a new page is great, but missing the option to create new posts!?

 => Thanks, will look at it.

> help: if you made a site for clients you might want the client to contact you.

 => I see no problem here, Its help and documentation, not support.

> my account: if a client doesn´t have an account this could be irritating?

 => Good point, needs a little thinking.

> Depended Styles for developers + Inline Code Documentation:

 => The main keyword here is "Inline" ex: https://github.com/pojome/elementor/blob/release/1.9.0/includes/managers/elements.php#L223

@Sarah-p 
The Old view page button was there because every change you made and saved would overwrite the published version. Now you are working on a draft, which is the main difference. Also, you can always see the current version you have in the editor by going to history >> revisions >> current:
![w8u1aj8](https://user-images.githubusercontent.com/660572/34514397-f355f85a-f075-11e7-9b00-ea12fc2fadb2.png)

@anandau 
I saw you got your answer on Facebook already :)

Thanks everyone for reporting.

**Comment by bainternet** [comment]
## v1.9 Beta Version RC2

We just released a new beta version, please update your version to v1.9.0 RC2.
https://downloads.wordpress.org/plugin/elementor.1.9.0-RC2.zip


This beta release includes:
- Fixed auto revisions created for a draft (thanks @wsydney76  for reporting).
- Fixed PHP notice when revisions is disabled via `define( 'WP_POST_REVISIONS', false );`
- Fixed Responsiveness of Template Library for small screens (thanks @tscheipi for reporting).
- Aligned with `AUTOSAVE_INTERVAL` for Autosave.
- Removed "Discard & Exit" button under Save options.
- Added Accordion Widget controls: 
   - Title HTML tag.
   - Title padding.
   - Content padding.
   - Icon and Active Icon.
   - Icon alignment.
   - Icon, Icon active colors.
   - Icon spacing.

- Added Toggle widget Controls:
   - Title HTML tag.
   - Title padding.
   - Content padding.
   - Icon and Active Icon.
   - Icon alignment.
   - Icon, Icon active colors.
   - Icon spacing.
   - Space between toggle Items.
   - Toggle Item box shadow.

The bottom line of this RC is that we have heard your cry and you can now "Disable" the autosave feature like you normally would for the native editor by settings the interval to a day or a year ex:
`define('AUTOSAVE_INTERVAL', 86400);`

**Comment by SniffleValve** [comment]
"Removed "Discard & Exit" button under Save options."

OK, maybe not enough coffee yet, but how am I now able to exit to the Dashboard? The only thing I could find is go all the way up to the hamburger menu and select Exit To Dashboard from there. Please tell me this isn't the way it's going be!  :)

**Comment by SniffleValve** [comment]
"Added Accordion Widget controls:"

Missing one ... All Closed on Page Load ;)

**Comment by arielk** [comment]
@SniffleValve The exit option is still 2 clicks away..

**Current:** X button > Go to Dashboard

**New:** Menu button > Exit to Dashboard

I think it makes more sense to separate between the Save/Update options and the Exit option.

**Comment by drank** [comment]
@arielk 
I think that the Exit function should be more accessible than it is now. With both the old and the new behaviors one has to click too many times just to exit Elementor. Can you just put a simple exit button in the bottom row, replacing the settings button, and move the settings to the hamburger at the top?
I would even like to request a keyboard shortcut for exiting Elementor that would directly exit me to the Library (if I started from there) or to Pages (if I started from there) and so on, and not only to the current page editing view (I never need this view when I exit Elementor, I always have to navigate one more time in order to get to the Elementor library or the Pages).

**Comment by JourneysCT** [comment]
Must say, I'm with SniffleValve. I would very much like to see the Accordion Widget with All closed on Page Load. Also, the Toggle Widget is the same way. There seems to be quite a few people asking for this to be fixed.

I'm in agreement as well with needing a better way to Exit Elementor. I think an X in the bottom left (Like Before) is much easier to utilize. Even if it's still 2 clicks, everyone is already familiar with such a necessary thing.

**Comment by black-eye** [comment]
Agreed with @SniffleValve, @JourneysCT and @drank on exit button - the former place was better. Just because it is visible righ away and everyone is used to it there. I can imagine customers calling me, "how the hack do I exit Elementor now" - even I was a bit confused, and I am a developer. ;-)

Also like the idea of the direct exit to Pages, Library or Posts (or whatever).

**Comment by black-eye** [comment]
I'm also not sure about the popup control of typography and text shadow. What was wrong with the curent solution? Now it's a bit buggy (try to open them both).

And when I click on the text shadow icon - it's immediatelly applied, which is a bit annoying. Sometimes I just qickly scanning through the options, clicking on the icon by mistake (or to check the settings) and after a while I'm wondering, why the hack my text has a black shadow...

**Comment by drank** [comment]
Something strange - what is the purpose of the arrow icon next to "Typography" and "Text Shadow"?
I thought it was for reverting to the default settings but it is not what happens when I click it.
Also, why is it shown (if the purpose is to revert to defaults) only when I click on the edit Typography button - I haven't edited nothing yet! When I click the arrow, it just closes the dialog.

I also think that the old way of showing Typography (and other) controls was better. If you wanted to remove the clutter, could you use an accordion type of grouping and not a pop-up?

**Comment by wsydney76** [comment]
I agree that the typography etc controls are not so intuitive. Maybe there should be some tooltips explaning them.

Personally i would like to see the old on/off switch back, but with a different behavior: if you switch it on, the popup appears immediatly, if it is already on, an edit link/button should be next to it to open the popup.

**Comment by Sarah-p** [comment]
I too agree that the old way for text/text-shadow was fine. I prefer accordion to pop-ups and flyouts. Since there are the only controls that do this it's not a consistent experience.

If you're going to have accordion/toggle title active color, you should probably add an active color in for title background as well. And if while you're add it give Tabs an active background color as well. And I agree there should be an all close or all open option

As for the dashboard widget, I would like to be able to set the title first, before going into Elementor. It's not intuitive that you then need to go to settings to set the title. Unless you can get the settings screen to open first when you use the dashboard widget.

Also now there is no way to get to the page edit screen, dashboard(if something says dashboard then it should be the dashboard, not the page edit screen), or a live version of the page under update. Having to go to the hamburger menu is not intuitive. I still think the eye icon should be for a live page not a preview since the edit screen allows you to preview anyway. Even the wp default editor has a 'view page' link. Now we have to go all the back to the dashboard then view page.

**Comment by avermat** [comment]
**Typography Popup**
I like the idea of the Typography popup, however, a few things make it really hard to understand in it's current form.

1) I can't close the popup by clicking the blue pencil icon. Nothing happens. I have to click in the white area around the pencil in order to close it.

2) What does the refresh/undo icon do? Refresh to default? Undo my last change? Undo all changes? Turn typography off?

3) How do I turn the typography off? The pencil turns it on, but the pencil does not turn it off?

It took me a long time to figure out how things worked. I think a much better solution would be to have an on/off button and an pencil/edit button. It would make way more sense. Then theres no confusion or learning curve to figure out what is actually going on. **One button turns typography on and off and one button edits the settings.** The first time you turn it on, the pencil popup would be triggered so I don't have to click twice. Then in the future, if I want to edit the settings I click the pencil. If I want to turn it off, I click the off toggle.

**Template Search**
Love the template search, thank you for adding this to the My Templates section! Looking forward to image previews in My Templates, and also making it the default when it opens.

**Sidebar Hamburger Menu**
For some reason the hamburger menu turns into X on mouse-over. This seems really confusing. It almost feels like it's going to close the sidebar. I would suggest leaving it a hamburger on mouse-over just like you do with the other 9-block button on the right-hand side of the sidebar. I do like that back arrow a lot though, I think that's an improvement.

**Comment by SniffleValve** [comment]
@arielk
It may not seem like much but believe me, it is now a major "distance to travel" to get all the way up to the hamburger (Menu button - which BTW, mysteriously turns into and X when clicked???) and then "all the way down" to the Exit To Dashboard. Do that a hundred or so times a day and it get tedious FAST, not to mention exacerbating my  carpal tunnel syndrome  :) The current version I could do in one small movement to the X. I'm on a 24" monitor and the new way is a loooong way up there :)

**Comment by copen8** [comment]
Very good work with the template search options. I think many users are looking forward to use it 👍 

**But**
Agree in the case of the exit button. I don't really see any advantage with the new placement. In my opinon the new placement is not intuitive instead  confusing.  An exit-button should be easy to identify that is not the case with the new position.

**Comment by purbeckpixels** [comment]
After editing a page designated as Front Page (ID 2) with the latest RC, the (rather oddly situated) 'Exit to Dashboard' button returns me to a Custom Post Type also called front page (ID 2106) with the slug /customposttypename/front-page, and not the page I was editing (ID 2).

Update: This behaviour occurs in version 1.8.12 too. And changing the slug of the CPT makes no difference, it exits to ID 2106 every time with and without Gutenberg activated.

**Comment by purbeckpixels** [comment]
This is related to the global post issue I raised when it was (re)introduced in version 1.6.x and [referenced here](https://github.com/pojome/elementor/issues/2149). It hasn't been fixed yet, in either 1.8.12 or 1.9 RC2 (I just tested it by using the The Events Calendar Shortcode plugin, which is where I [first noticed this behaviour](https://github.com/pojome/elementor/issues/2077#issuecomment-318915347). Versions 1.6.x > 1.9 and it still hasn't been fixed (it briefly was during the RC phase of 1.6, though).

**Comment by bainternet** [comment]
@purbeckpixels 

Checking the code we of the "Exit to Dashboard URL" https://github.com/pojome/elementor/blob/release/1.9.0/includes/editor.php#L534 the Post ID is set  for Editor so the only place it can change is by filter either `get_edit_post_link` or `elementor/utils/exit_to_dashboard_url` which are both triggered by [`get_exit_to_dashboard_url `](https://github.com/pojome/elementor/blob/release/1.9.0/includes/utils.php#L179).

So please check if you have something hooking into these, if not please contact our support and see how they can help (support@elementor.com)

**Comment by purbeckpixels** [comment]
It's related to the The Events Calendar Shortcode plugin and other plugins that change the global post. If I deactivate the TECS plugin everything works as it should - reactivate it and the issues return. I picked up on this months ago and you managed to fix it in 1.6 RC4 ([see here](https://github.com/pojome/elementor/issues/2077#issuecomment-318915347)), Versions 1.6 and 1.6.1 worked fine, but the 'bug' returned in 1.6.2 and has been present ever since.

The TECS plugin is relatively simple, but [line 140](https://plugins.trac.wordpress.org/browser/the-events-calendar-shortcode/trunk/the-events-calendar-shortcode.php) -  `global $wp_query, $post;` - seems to be the cause - if I comment it out and reactivate the plugin, the plugin outputs the front/home page as an event and the 'Exit it to Dashboard' URLs link to the wrong WP edit screens. Uncomment out line 140 and the correct event is displayed on the front end, but the 'Exit to Dashboard' URLs still link to the wrong WP edit screens. I've read somewhere that $post global is unreliable and can be easily changed by other functions, which seems to be the case here.

**Comment by purbeckpixels** [comment]
Replacing `wp_reset_query();` on [line 368](https://plugins.trac.wordpress.org/browser/the-events-calendar-shortcode/trunk/the-events-calendar-shortcode.php) of the The Events Calendar Shortcode plugin with `wp_reset_postdata();` fixes the issue.

`wp_reset_query();` is only required after using `query_posts()`. The TECS plugin uses `$wp_query` on [line 140](https://plugins.trac.wordpress.org/browser/the-events-calendar-shortcode/trunk/the-events-calendar-shortcode.php), which is the global instance of the WP_Query class, thus `wp_reset_postdata();`should be used instead. From the WordPress Codex:

> Calling wp_reset_query is not necessary after using WP_Query or get_posts as these don't modify the main query object. Instead use wp_reset_postdata.

So not Elementor's fault.

**Comment by purbeckpixels** [comment]
The The Events Calendar Shortcode plugin developer has fixed the issue and it's already available as an update in WordPress.

I've only been struggling to get to the bottom of this since last July... :)

**Comment by purbeckpixels** [comment]
Has lightbox functionality been updated to work with Gutenberg yet? It's stopped working on the dev site I'm testing RC2 on with Gutenberg, so I'm guessing not.

**Comment by maxwellhibbert** [comment]
Global widgets aren't saving for me when the page is saved.

**Comment by bainternet** [comment]
## v1.9 Beta Version RC3

We just released a new beta version, please update your version to v1.9.0 RC3.
https://downloads.wordpress.org/plugin/elementor.1.9.0-RC3.zip

Thanks, everyone for helping us improve, again we try to listen as much as possible and in this RC we tried to make the Group Control Popup more intuitive by fixed the "Not Closing" bug and adding a label/tooltip to make it more understandable.

This beta release includes:
- Fixed close the "Group Control popup" by clicking the blue pencil icon (thanks, @avermat)
- Added "Back to default" tooltip Group Control "Default" button. 
![smvcrga](https://user-images.githubusercontent.com/660572/34675127-41979a72-f491-11e7-9a8f-88857090b476.png)

- Fixed Global Widget not saving (thanks, @maxwellhibbert)
- Removed "My Account" (pro)  button from Overview Dashboard widget (thanks, @tscheipi)

If you have any bugs Not related to this RC please Feel free to open an issue and keep this RC related only.

Thanks again to everyone.

**Comment by drank** [comment]
To be honest, I am little bit disappointed by this release - how come adding a "Back to default" tooltip is considered a fix, this is more of a patch for something that should be intuitive enough without any tooltips. You are programmers, you know the rule - "the code should speak for itself, weak code requires many comments". Why did you need to ditch the On/Off control for the Typography in the 1st place. It was working great. A better improvement would be to make it expand (not popup like now) when set to On - just like in modern applications (e.g. iOS or Android - check your settings on your phones - there are no shaky popups there). The popup is ugly, it obscures the view underneath it and bringing it was not dictated by any UX necessity (IMHO).

Sorry for sounding so harsh - I love Elementor very much and I would like to see it improve and stabilize, not crumble upon its own "importance" ;)

But I understand you are rushing towards the promised release on 9th. Can you postpone and polish? I think this would be for the better :)

Wish you luck!

Best regards,
Drago

**Comment by black-eye** [comment]
I have to agree with @drank on the Typo and Text Shadow Popup - I find it a bit tricky and unintuitive, especially conciderng the fact, that these are the only 2 popups in all the widgets (but maybe, more are coming).

I choose Elementor because of it's concern on UX experience an so far I wasn't dissapointed (at all). But this release brings some UX changes, that seem to me as a little step backwards: popups and new placement of the exit button.

**Comment by avermat** [comment]
1) I think the popups are a real improvement. It really cleans up the sidebar and makes it easier to navigate when a lot of typography/box shadow/etc options are turned "on". I used to get lost in all of the options within certain widgets that had a lot of settings.

2) The "back to default" tooltip is an improvement. However, I have to agree with others that it's still unintuitive. How would I know that clicking the pencil turns the setting "on"? How would I know that a blue pencil means "on"? Using the "on/off" switch is sooo much easier to understand. Why not just add the on/off switch next to the pencil button instead of using "back to default"? 

3) In all of the Internet I have never seen a hamburger turn into an X on mouse-over. I can't see how this makes sense. Hamburgers open menus, they don't close things.

This latest release was a little alarming. If people in the github group find this stuff confusing I can only imagine how new users will struggle with this. I don't think this is a case of people resisting change, but rather that some of these additions are truely unintuitive in comparison to the old version.

Thank you again for considering our feedback. All the best.

**Comment by bainternet** [comment]
Thanks everyone for all the help, feedback and suggestions.

Last change we managed to push in is removing the X when you hover on the hamburger menu (thanks, @avermat )

[Introducing Elementor v1.9: Never Lose Momentum With Autosave](https://elementor.com/autosave/)

Full changelog: 1.9.0 – 2018-01-09

- New: Added Autosave capability for editor
- New: Added save as draft option
- New: Brand new Template Library
- New: Added filter & sorting (new, trend & popular) to Template Library (#1711)
- New: Added import & sync tools to Template Library (#2402)
- New: Added search form to Template Library (#2499)
- New: Added my favorites option to the Template Library
- New: Added sorting (name, type, author & date) to My Templates Library
- New: Added new social icons for Meetup, RSS & Skype (#2703, #2701, #3090)
- New: Added Overview dashboard widget
- Tweak: Added privacy control to video widget (#2741)
- Tweak: Added new controls for Icons, Spacing & Padding to Toggle widget (#2836)
- Tweak: Added new controls for Icons & Padding to Accordion widget (#2836)
- Tweak: Added responsive space between control to Icon List widget
- Tweak: Rename “Page Settings” panel to “Document Settings”
- Tweak: Moved My Library to top of admin menu (#2843)
- Tweak: Re-design menu of editor panel
- Tweak: Replaced jquery-simple-dtpicker with flatpickr JS library (#2095)
- Tweak: Reduced panel clutter by allowing group control to be displayed as a popup
- Tweak: Added Gutenberg compatibility – “Add New Elementor” and “Back to Classic” edit screen
- Tweak: Added excerpt control to the Document Settings if the post type supports excerpts (#2837)
- Tweak: Added text-decoration to Typography control
- Tweak: All functions, action hooks and filter hooks have been documented
- Tweak: Implementing inline editing functionality in various Elementor widgets
- Tweak: Accessible widget search – add label for screen readers in the search box (#2835)
- Tweak: Improved Editor accessibility – replace title attributes with aria-label (#2861)
- Tweak: Progress Bar widget accessibility – set correct role and added new aria attributes (#2861)
- Tweak: Alert widget accessibility – make the “Dismiss” button accessible (#3108)
- Tweak: Tabs widget accessibility – added id, role and aria attributes (#2913)
- Tweak: Toggle widget accessibility – added id, role and aria attributes (#2836)
- Tweak: Accordion widget accessibility – added id, role and aria attributes (#2836)
- Tweak: Social Icons widget accessibility – added labels for screen readers
- Tweak: Added Browser support notification for unsupported browsers
- Tweak: Depended styles, various elements can set stylesheet dependencies
- Tweak: Added option to set control as required by required => true
- Fix: Added wptexturize filter to match WordPress native text formatting
- Fix: Alignment issue with Icon widget

and again many thanks to everyone who tested and helped improve it.

**Comment by scottwyden** [comment]
I updated and on WP Engine the backend times out to a 502 error. When I disable Elementor, all is right in the world.  I checked with them if there are any errors in the logs, but apparently:

"I am not seeing anything that stands out. Just a sharp dip in performance. You may want to install query monitor (plugin) and update or activate Elementor again, and see if there a specific conflict with any queries being made by that plugin."

**Comment by gitapanhar** [comment]
Where's the template library option? I cant open the template library.

**Comment by JohnK75** [comment]
Under element style > entrance animation... when this is enabled on my typography, after the animation plays, the typography disappears and I can no longer edit the type. Is this a known bug, and/or is there a current suggested fix?

**Comment by ianstudio** [comment]
I find popups truly annoying actually.

@arielk I proposed the Elementor team allows the toolbox to be paced on the right if so desired (as is possible with the new Visual Composer Website Builder (I'm really looking into.. for future projects). When right-handed (most people..) it'd already be a little help in the traveling mouse department and carpal tunnel syndrome issue.. these kids cannot understand yet.

**Comment by NatnaelFikru** [comment]
where is the file dud! shit!

**Comment by soyket** [comment]
@NatnaelFikru look at the beginning!
